tools/sotn_build/get_file_list.py

```diff
@@ -0,0 +1,311 @@
+    asmPath: str = options["asm_path"].rstrip("/")
+    dataPath: str = os.path.join(options["asm_path"], "data").rstrip("/")
+
+    asmFiles: set = {
+        FILENAME_TEMPLATE.substitute(
+            path=asmPath, filename=seg[2] if len(seg) >= 3 else seg[1], ext="s"
+        )
+        for seg in segments
+        if len(seg) >= 2 and seg[1] in SEG_TYPES["asm"]
+    }
+
+    # A seg type is added with a leading . for each existing data seg type
+    if not options["ignore_hidden"]:
+        dataSegTypes = SEG_TYPES["data"] + tuple(
+            f".{type}" for type in SEG_TYPES["data"]
+        )
+
+    dataFiles: set = {
+        FILENAME_TEMPLATE.substitute(
+            path=dataPath,
+            # If a segment has 3 or more elements, then it has a name
+            # If it only has 1 or 2 elements, we use the memory address as the name
+            filename=seg[2] if len(seg) >= 3 else f"{seg[0]:X}",
+            # If it is only 1 element, then we assume it is bss and that the element is a memory address
+            ext=f"{seg[1].lstrip('.')}.s" if len(seg) >= 2 else "bss.s",
+        )
+        for seg in segments
+        if (len(seg) >= 2 and seg[1] in (dataSegTypes)) or len(seg) == 1
+    }
+
+    return {file for file in dataFiles | asmFiles if os.path.exists(file)}
+
+
+def GetSrcFiles(options: dict, segments: list[list]) -> set[str]:
+    """Constructs a list of source files from splat segments"""
+    srcPath: str = options["src_path"].rstrip("/")
+
+    # We assume that if there is a data file, then there should be a c file for it
+    # This is safe since we're using sets and we validate existence on the final set
+    extraFiles: set = {
+        FILENAME_TEMPLATE.substitute(path=srcPath, filename=seg[2], ext="c")
+        for seg in segments
+        if len(seg) >= 3 and seg[1].startswith(".")
+    }
+
+    cFiles: set = {
+        FILENAME_TEMPLATE.substitute(
+            path=srcPath,
+            filename=seg[2] if len(seg) >= 3 else f"{seg[0]:X}",
+            ext=seg[1],
+        )
+        for seg in segments
+        if len(seg) >= 2 and seg[1] in SEG_TYPES["src"]
+    }
+
+    if options["include_shared"]:
+        srcPath = srcPath.replace(f"/{options["ovl"]}", "")
+        cFiles.update(
+            {
+                os.path.join(srcPath, file)
+                for file in os.listdir(srcPath)
+                if file.endswith(".c")
+            }
+        )
+
+    return {file for file in cFiles | extraFiles if os.path.exists(file)}
+
+
+# This function still needs a bit of refinement, but it is serviceable for now
+def GetAssetFiles(
+    options: dict, segments: list[list], assetFilters: tuple[tuple]
+) -> set[str]:
+    """Constructs a list of asset files using a directory listing or the segments"""
+    assetPath = options["asset_path"].rstrip("/")
+
+    # This conditional needs to be cleaned up and probably relocated for easier code maintenance
+    if (
+        options["platform"] == "psx"
```

Comment:
Should this be `us`?

Reply:
This applies to both us and hd (they both use psx as platform in the splat configs). Really just a different way of saying "not psp".